host.py

```diff
@@ -294,11 +327,10 @@ def port_has_carrier(self, port_name: str) -> bool:
         return "NO-CARRIER" not in ports[port_name]["flags"]
 
     def write(self, fn, contents):
-        if self.is_localhost():
-            with open(fn, "w") as f:
-                f.write(contents)
-        else:
-            raise Exception("Not implemented")
+        with open(fn, "w") as f:
+            f.write(contents)
```

Comment:
tricky to write it as a tmp file locally with the same name. if writing to remote host, can we create a tmpfile (something along the lines of mktmp or some similar function) and scp (or sftp) that one? We will avoid quite some issues later on if we use that instead.